src/optimalcontrol/include/iDynTree/Integrators/RK4.h

```diff
@@ -30,23 +30,24 @@ namespace iDynTree {
          * \ingroup iDynTreeExperimental
          */
 
-            class RK4 : public Integrator
+            class RK4 : public FixedStepIntegrator
             {
                 Eigen::SparseMatrix<double> m_aCoefficents;
                 Eigen::VectorXd m_bCoefficients;
                 Eigen::VectorXd m_cCoefficients;
                 Eigen::MatrixXd m_K;
                 VectorDynSize m_computationBuffer;
 
-                bool oneStepIntegration(double t0, double dT, const VectorDynSize& x0, VectorDynSize& x);
+                bool oneStepIntegration(double t0, double dT, const VectorDynSize& x0, VectorDynSize& x) override;
 
-            public:
-                RK4(const std::shared_ptr<iDynTree::optimalcontrol::DynamicalSystem> dynamicalSystem);
+                bool allocateBuffers() override;
 
-                ~RK4();
+            public:
+                RK4();
 
-                bool integrate(double initialTime, double finalTime) override;
```

Comment:
If `integrate()` is not implemented, should it return false instead of not being overridden?

Reply:
The method ``integrate`` is implemented in ``FixedStepIntegrator``.